components/jaggery-core/org.jaggeryjs.jaggery.core/src/main/java/org/jaggeryjs/jaggery/core/manager/CommonManager.java

```diff
@@ -301,35 +306,47 @@ public static void print(Context cx, Scriptable thisObj, Object[] args, Function
         OutputStream out = (OutputStream) jaggeryContext.getProperty(CommonManager.JAGGERY_OUTPUT_STREAM);
         if (args[0] instanceof StreamHostObject) {
             InputStream in = ((StreamHostObject) args[0]).getStream();
+            ReadableByteChannel inputChannel = null;
+            WritableByteChannel outputChannel = null;
+            FileChannel fc = null;
             try {
-                byte[] buffer = new byte[BYTE_BUFFER_SIZE];
-                int count;
-                while ((count = in.read(buffer)) != -1) {
-                    out.write(buffer, 0, count);
-                }
-                in.close();
+                inputChannel = Channels.newChannel(in);
+                outputChannel = Channels.newChannel(out);
+                // copy the channels
+                fc = (FileChannel) inputChannel;
+                fc.transferTo(0, fc.size(), outputChannel);
             } catch (IOException e) {
                 log.debug(e.getMessage(), e);
                 throw new ScriptException(e);
             } finally {
-                if (in != null) {
-                    try {
-                        in.close();
-                    } catch (IOException e) {
-                        log.debug(e.getMessage(), e);
-                    }
-                }
+                clearResources(fc, inputChannel, outputChannel, in, out);
             }
         } else {
             try {
                 out.write(HostObjectUtil.serializeObject(args[0]).getBytes());
             } catch (IOException e) {
-                log.debug(e.getMessage(), e);
+                if (log.isDebugEnabled()) {
+                    log.debug(e.getMessage(), e);
+                }
                 throw new ScriptException(e);
             }
         }
     }
 
+    private static void clearResources(Closeable... resources) {
+        for (Closeable resource : resources) {
+            if (resource != null) {
+                try {
+                    resource.close();
+                } catch (IOException e) {
+                    if (log.isDebugEnabled()) {
```

Comment:
Already fixed in [#4ecf8fe](https://github.com/wso2/jaggery/commit/4ecf8feb00439c424fa95803bbf95b2f8a922b61).